crates/sui-indexer-alt/src/pipeline/logging.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::time::Instant;
+
+use tracing::{debug, info};
+
+use super::Processor;
+
+/// Tracing message for the watermark update will be logged at info level at least this many
+/// checkpoints.
+const LOUD_WATERMARK_UPDATE_INTERVAL: i64 = 5 * 10;
+
+pub(crate) struct WatermarkLogger {
+    name: &'static str,
+    timer: Instant,
+    prev_checkpoint: i64,
+    prev_transaction: Option<i64>,
+}
+
+impl WatermarkLogger {
+    pub fn new(name: &'static str, init_checkpoint: i64, init_transaction: Option<i64>) -> Self {
+        Self {
+            name,
+            timer: Instant::now(),
+            prev_checkpoint: init_checkpoint,
+            prev_transaction: init_transaction,
+        }
+    }
+
+    pub fn log<H: Processor>(
```

Comment:
nit: doc comment -- particularly to explain how the TPS/CPS metrics are calculated (the detail about it sampling between load updates).